pkg/util/mount/mount.go

```diff
@@ -108,6 +108,11 @@ type Interface interface {
 	// subpath starts. On the other hand, Interface.CleanSubPaths must be called
 	// when the pod finishes.
 	PrepareSafeSubpath(subPath Subpath) (newHostPath string, cleanupAction func(), err error)
+	// KubeletPath returns the path name kubelet can access, whether
+	// kubeleth is running in a container or not.
+	// Note: path on the host may be different to the path kubelet can access
+	// because in container rootfs of the host is not mounted at OS root "/".
+	KubeletPath(pathname string) (string, error)
```

Comment:
What if instead of exposing KubeletPath as a new interface, and then directly calling os functions, we add the operation we need into the mounter interface? In this case, get fsgroup?

Reply:
What I mean is, can kubelet path be an internal method to nsenter?  And getfsgroup, getmountrefs can be mounter interfaces?  I think it is better if volume plugin layer doesn't have to worry about path manipulation or environment

Reply:
@msau42 Yes, it can be. I planned to do refactoring in https://github.com/kubernetes/kubernetes/pull/62102. Moving `KubeletPath` to nsenter is a good idea! Done in https://github.com/kubernetes/kubernetes/pull/62903/commits/6508abd8713ea3a389ae817463d4c431cadbf50e. local.go is [much better now](https://github.com/kubernetes/kubernetes/pull/62903/commits/6508abd8713ea3a389ae817463d4c431cadbf50e#diff-ec924773446516c2e7bf84c477f5f4dd).